### ISSUE TITLE
Underscore attribute ruins initiating an instance from dict data

### ISSUE THREAD

**Issue by MacHu-GWU** [python-attrs/attrs#263]
```python
@attr.s
class User(object):
    _id = attr.ib()
    name = attr.ib()

doc = {"_id": 1, "name": "Alice"}
```
when working with mongodb, how could I create instance for this from mongodb document?

because we know that ``User(**doc)`` doesn't work due to the underscore.

**Comment by Tinche** [comment]
Well there are many ways. You could try this:

```
@attr.s
class User(object):
    _id = attr.ib()
    name = attr.ib()

    @classmethod
    def from_mongo(cls, doc):
        return cls(**{k if not k.startswith('_') else k[1:]: v for k, v in doc.items()})
```

**Comment by hynek** [comment]
Questions like this are being asked basically once week now.  I’m afraid I’ll have to write a lot more docs.

**Comment by hynek** [comment]
fixed in #370
